### ISSUE TITLE
0.1.29 doesn't build against musl libc.

### ISSUE THREAD

**Issue by sfackler** [rust-lang/jobserver-rs#79]
For example, building in an alpine container:

```
6.767 error[E0425]: cannot find function `preadv2` in crate `libc`
6.767     --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/jobserver-0.1.29/src/unix.rs:598:15
6.767      |
6.767 598  |           libc::preadv2(
6.767      |                 ^^^^^^^ help: a function with a similar name exists: `preadv`
6.767      |
6.767     ::: /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.153/src/unix/linux_like/linux/mod.rs:4868:13
6.767      |
6.767 4868 | /             pub fn preadv(
6.767 4869 | |                 fd: ::c_int,
6.767 4870 | |                 iov: *const ::iovec,
6.767 4871 | |                 iovcnt: ::c_int,
6.767 4872 | |                 offset: ::off_t,
6.767 4873 | |             ) -> ::ssize_t;
6.767      | |__________________________- similarly named function `preadv` defined here
6.767
6.768 error[E0425]: cannot find value `RWF_NOWAIT` in crate `libc`
6.768     --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/jobserver-0.1.29/src/unix.rs:606:19
6.768      |
6.768 606  |             libc::RWF_NOWAIT,
6.768      |                   ^^^^^^^^^^ help: a constant with a similar name exists: `IPC_NOWAIT`
6.768      |
6.768     ::: /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.153/src/unix/linux_like/linux/mod.rs:2102:1
6.768      |
6.768 2102 | pub const IPC_NOWAIT: ::c_int = 0o4000;
6.768      | ----------------------------- similarly named constant `IPC_NOWAIT` defined here
6.768
6.821 For more information about this error, try `rustc --explain E0425`.
```

https://docs.rs/libc/latest/x86_64-unknown-linux-musl/libc/index.html?search=preadv2

**Comment by eeeebbbbrrrr** [comment]
Dunno if this is helpful but I see the same using Docker images based on Alpline-3.12 and Alpine 3.19.

**Comment by rdbo** [comment]
Yep, having the same issue here as well, on x86_64 Alpine edge.

**Comment by Asempere123123** [comment]
Same issue on rust:alpine3.19 docker image

**Comment by weihanglo** [comment]
Looks like musl libc has gained support of `preadv2` since 1.2.5: <https://www.openwall.com/lists/musl/2024/03/01/2>, though that might be too new and the latest [`libc`](https://crates.io/crates/libc/0.2.153) doesn't have that at this moment.

**Comment by randomairborne** [comment]
Also having this issue.

**Comment by weihanglo** [comment]
> Also having this issue. Perhaps MUSL compatibility should also be tested in CI?

Feel free to open a PR for that if it is not too complicated to set up in CI.

**Comment by weihanglo** [comment]
https://github.com/rust-lang/jobserver-rs/pull/80#issuecomment-2052112449

> Published 0.1.30 and yanked 0.1.29.
